libnavigation-router/build.gradle

```diff
@@ -61,6 +45,10 @@ dependencies {
     apply from: "${rootDir}/gradle/unit-testing-dependencies.gradle"
 }
 
+task kdocJar(type: Copy) {
+    // empty task, no docs for current module
```

Comment:
If in the future this module adds any public documentation, we'd need to bring `dokkaHtml { }` and `apply plugin: 'org.jetbrains.dokka'` back and add `from dokkaHtml.outputDirectory` here. Would it be better leaving it as is and if there's nothing to copy docs aren't generated for that module? 🤔

Reply:
Looking at it again, I don't personally mind that we're producing javadocs/kdocs for internal classes, the code is public so we might as well append the inline docs for private/internal artifacts if there are any.

The problem is that we're creating directories with those internal docs on the web page. Maybe we should leave the old setup as is, and instead, add a script that removes the internal libraries from being published to the web page to reduce noise.

Reply:
How about we edit generated files ourselves with a script to remove those directories and links?

Reply:
Yeah something around these lines https://github.com/mapbox/mapbox-navigation-android/pull/3890#discussion_r551996117 is what I was thinking as well.

Reply:
in that case we depend on dokka's generated html file and it might be changed in any version. IMO setup dokka/gradle better than fix output manually

Reply:
Couldn't we check in the `androidKdocJar` Gradle task? If empty, docs aren't copied. Is that an option?

Reply:
@Guardiola31337 not sure I follow how to make it, could you give a sample?

Reply:
Would need to investigate how to do it in Gradle / Groovy but probably it is an option worth looking.

Reply:
if we keep legacy implementation and keep dependsOn `dokkaHtml` we force every single module uses `dokka` plugin. That's not what we want

Reply:
That shouldn't a problem as long as we detect there are no docs so they're not copied over. Does that make sense? Should be feasible in Gradle / Groovy.

Reply:
legacy task depends on `dokkaHtml`, we should remove it. After that we should detect docs folder, right? 
```
task androidJavadocsJar(type: Jar, dependsOn: dokkaHtml) {
```